src/app/social-media/social-media-input/social-media-input.component.ts

```diff
@@ -0,0 +1,142 @@
+import { Component, OnInit, AfterViewInit, OnDestroy } from '@angular/core';
+import { Router, ActivatedRoute } from '@angular/router';
+import { SocialMediaType } from '../../view-models/social-media-type';
+import { SocialMediaPostExtended } from '../../view-models/social-media-post-extended';
+import { NavmenuService } from '../../services/navmenu.service';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { DeletePostConfirmationModalComponent } from '../delete-post-confirmation-modal/delete-post-confirmation-modal.component';
+import { AddSocialMediaPostModalComponent } from '../add-social-media-post-modal/add-social-media-post-modal.component';
+import { SocialMediaPostsService } from '../../services/socialMediaPosts.service';
+import { SocialMediaRenderService } from '../../services/socialMediaRender.service';
+
+@Component({
+  selector: 'app-social-media-input',
+  templateUrl: './social-media-input.component.html',
+  styleUrls: ['./social-media-input.component.scss']
+})
+
+export class SocialMediaInputComponent implements OnInit, AfterViewInit, OnDestroy {
+  socialmedia: SocialMediaPostExtended[];
+  selectedSocialMedia: SocialMediaPostExtended[];
+
+  socialmediatypes: SocialMediaType[];
+  filterBySocialMediaType: string;
+
+  constructor(
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    public nav: NavmenuService,
+    private modal: NgbModal,
+    private socialMediaService: SocialMediaPostsService,
+    private socialMediaRenderService: SocialMediaRenderService) {
+    this.router.routeReuseStrategy.shouldReuseRoute = () => false;
+  }
+
+  ngOnInit() {
+    this.nav.hide();
+    this.activatedRoute.data.subscribe(data => {
+      this.socialmedia = data['socialmedia'];
+    });
+    const selectedSocialmediatypes = [];
+
+    this.socialmedia.forEach(post => {
+      if (selectedSocialmediatypes.indexOf(post.mediaType) === -1) {
+        selectedSocialmediatypes.push(post.mediaType);
+        this.socialMediaRenderService.loadWidgets(post.mediaType);
+      }
+    });
+  }
+
+  ngAfterViewInit() {
+    const selectedSocialmediatypes = [];
+    if (this.socialmedia !== null || this.socialmedia.length > 0) {
+      this.socialmedia.forEach(post => {
+        if (selectedSocialmediatypes.indexOf(post.mediaType) === -1) {
+          selectedSocialmediatypes.push(post.mediaType);
+          this.socialMediaRenderService.loadWidgets(post.mediaType);
+        }
```

Comment:
It seems like both OnInit and AfterViewInit do the same thing. Maybe this can be abstracted out into another method they both call or does it even need to be done in both places?